test/lib/SchemaConnector-test.js

```diff
@@ -0,0 +1,301 @@
+  })
+  .stateRefreshHandler((accessToken, response) => {
+    response.addDevice('abcd', [
+      {
+        component: 'main',
+        capability: 'st.switch',
+        attribute: 'switch',
+        value: 'off'
+      }
+    ])
+  })
+  .commandHandler((accessToken, response, devices) => {
+    for (const device of devices) {
+      response.addDevice(device.externalDeviceId, device.commands.map(cmd => {
+        return {
+          component: cmd.component,
+          capability: cmd.capability,
+          attribute: 'switch',
+          value: cmd.command === 'on' ? 'on' : 'off'
+        }
+      }))
+    }
+  })
+  .integrationDeletedHandler(accessToken => {
+    delete accessTokens[accessToken]
+  });
+
+describe('SchemaConnector', function() {
+
+  describe('constructor', function() {
+    it('Should create an instance of StateUpdateRequest', async function() {
+      schemaConnector.should.be.instanceOf(SchemaConnector);
+    });
+  });
+
+  describe('discoveryHandlerRequest', function() {
+    it('Should return proper discovery response', async function() {
+      const response = await schemaConnector.handleMockCallback({
+        "headers": {
+          "schema": "st-schema",
+          "version": "1.0",
+          "interactionType": "discoveryRequest",
+          "requestId": "0edb967a-380e-4699-968e-64ea31cef618"
+        },
+        "authentication": {
+          "tokenType": "Bearer",
+          "token": "ACCT-HCtR4Q"
+        }
+      });
+
+      response.should.have.property('headers');
+      response.headers.schema.should.equal('st-schema');
+      response.headers.version.should.equal('1.0');
+      response.headers.interactionType.should.equal('discoveryResponse');
+      response.headers.requestId.should.equal('0edb967a-380e-4699-968e-64ea31cef618');
+
+      response.should.have.property('devices');
+      response.devices.length.should.equal(1);
+      response.devices[0].externalDeviceId.should.equal('abcd');
+      response.devices[0].friendlyName.should.equal('Test Switch 1');
+      response.devices[0].deviceHandlerType.should.equal('c2c-switch');
+    });
+  });
+
+  describe('grantCallbackAccess', function() {
+    it('Should return proper state refresh response', async function() {
+
+      let expectedAccessToken = 'UNDEFINED';
+      let expectedCallbackAuthentication = {};
+      let expectedCallbackUrls = {};
+
+      schemaConnector.callbackAccessHandler(async (accessToken, callbackAuthentication, callbackUrls) => {
+        expectedAccessToken = accessToken;
+        expectedCallbackAuthentication = callbackAuthentication;
+        expectedCallbackUrls = callbackUrls;
+      });
+
+      let tokenUrl = 'UNDEFINED';
+      let tokenCode = 'UNDEFINED';
+
+      const stub = sinon.stub(AccessTokenRequest.prototype, 'getCallbackToken').callsFake(async function(url, code) {
+        tokenUrl = url;
+        tokenCode = code;
+        return new Promise((resolve, reject) => {
```

Comment:
since this function is already async, you don't need to create a new instance of Promise here.

Reply:
:thumbsup: